From 9dcfc7daa4c940992920d3b2fbd317da20e44aad Mon Sep 17 00:00:00 2001
From: "C. Michael Pilato" <cmpilato@collab.net>
Date: Tue, 24 Jan 2017 15:39:56 -0500
Subject: [PATCH] Escape some raw path data before handing off to templates

* lib/viewvc.py
  (nav_path): Escape the 'name' property of navigation path components
    the same way we escape that of the 'root' path component.

Reported by: Thomas Gerbet <thomas.gerbet@enalean.com>
---
 lib/viewvc.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/viewvc.py b/lib/viewvc.py
index 84bdca90..922baa2e 100644
--- a/lib/viewvc.py
+++ b/lib/viewvc.py
@@ -980,7 +980,7 @@ def nav_path(request):
     path_parts.append(part)
     is_last = len(path_parts) == len(request.path_parts)
 
-    item = _item(name=part, href=None)
+    item = _item(name=request.server.escape(part), href=None)
 
     if not is_last or (is_dir and request.view_func is not view_directory):
       item.href = request.get_url(view_func=view_directory,
